io.asgardio.java.oidc.sdk/src/test/java/io/asgardio/java/oidc/sdk/HTTPSessionBasedOIDCProcessorTest.java

```diff
@@ -0,0 +1,141 @@
+        State state = new State("SampleState");
+        RequestContext requestContext = new RequestContext(state, nonce);
+
+        HttpSession session = mock(HttpSession.class);
+        mockedOIDCManagerFactory = mockStatic(DefaultOIDCManagerFactory.class);
+        when(DefaultOIDCManagerFactory.createOIDCManager(oidcAgentConfig)).thenReturn(defaultOIDCManager);
+        when(request.getSession()).thenReturn(session);
+        when(defaultOIDCManager.sendForLogin(request, response)).thenReturn(requestContext);
+
+        HTTPSessionBasedOIDCProcessor provider = new HTTPSessionBasedOIDCProcessor(oidcAgentConfig);
+        provider.sendForLogin(request, response);
+
+        verify(session).setAttribute(SSOAgentConstants.REQUEST_CONTEXT, requestContext);
+    }
+
+    @Test
+    public void testHandleOIDCCallback() throws SSOAgentException {
+
+        SessionContext sessionContext = new SessionContext();
+        RequestContext requestContext = new RequestContext();
+
+        HttpSession session = mock(HttpSession.class);
+        mockedOIDCManagerFactory = mockStatic(DefaultOIDCManagerFactory.class);
+        when(DefaultOIDCManagerFactory.createOIDCManager(oidcAgentConfig)).thenReturn(defaultOIDCManager);
+        when(request.getSession()).thenReturn(session);
+        when(request.getSession(false)).thenReturn(session);
+        when(session.getAttribute(SSOAgentConstants.REQUEST_CONTEXT)).thenReturn(requestContext);
+        when(defaultOIDCManager.handleOIDCCallback(request, response, requestContext)).thenReturn(sessionContext);
+
+        HTTPSessionBasedOIDCProcessor provider = new HTTPSessionBasedOIDCProcessor(oidcAgentConfig);
+        provider.handleOIDCCallback(request, response);
+
+        verify(session).setAttribute(SSOAgentConstants.SESSION_CONTEXT, sessionContext);
+    }
+
+    @Test
+    public void testLogout() throws SSOAgentException {
+
+        RequestContext requestContext = new RequestContext();
+        SessionContext sessionContext = new SessionContext();
+
+        HttpSession session = mock(HttpSession.class);
+        mockedOIDCManagerFactory = mockStatic(DefaultOIDCManagerFactory.class);
+        when(DefaultOIDCManagerFactory.createOIDCManager(oidcAgentConfig)).thenReturn(defaultOIDCManager);
+        when(request.getSession()).thenReturn(session);
+        when(request.getSession(false)).thenReturn(session);
+        when(session.getAttribute(SSOAgentConstants.SESSION_CONTEXT)).thenReturn(sessionContext);
+        when(defaultOIDCManager.logout(sessionContext, response)).thenReturn(requestContext);
+
+        HTTPSessionBasedOIDCProcessor provider = new HTTPSessionBasedOIDCProcessor(oidcAgentConfig);
+        provider.logout(request, response);
+
+        verify(session).setAttribute(SSOAgentConstants.REQUEST_CONTEXT, requestContext);
+    }
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+}
```

Comment:
Add a newline at the end of the file

Reply:
Resolved with https://github.com/asgardio/asgardio-java-oidc-sdk/pull/12/commits/8935b73e06c06ac1e1f216c89c0b19e175e1c196.